hudi-common/src/main/java/org/apache/hudi/common/table/view/HoodieTableFileSystemView.java

```diff
@@ -167,6 +167,12 @@ public HoodieTableFileSystemView(HoodieTableMetaClient metaClient, HoodieTimelin
     addFilesToView(fileStatuses);
   }
 
+  public HoodieTableFileSystemView(HoodieTableMetaClient metaClient, HoodieTimeline visibleActiveTimeline,
+      Map<String, FileStatus[]> partitionFiles) {
+    this(metaClient, visibleActiveTimeline);
+    addFilesToView(partitionFiles);
+  }
+
```

Comment:
On the reading side, FSView has to be aware of physical path. Statuses got passed in should be referring to the actual file. But this is breaking the logic of getting partition name from file path (See L